text/0096-contract-metadata.md

```diff
@@ -0,0 +1,206 @@
+# Casper On-Chain Contract Metadata Standard
+
+## Summary
+
+[summary]: #summary
+
+This standard proposes a unified way for Casper smart contracts to self-describe with on-chain metadata, improving contract discoverability and trust. It defines optional metadata fields – `name`, `description`, `icon_uri`, and `project_uri` – which a contract can expose via dedicated entry points. These fields allow a contract to broadcast its human-readable name and purpose, along with links to an icon image and project website.
+
+By adopting this standard, Casper developers enable wallets, network explorers, and indexers to fetch and display contract information directly from the blockchain, much like how token standards (e.g. CEP-18 for fungible tokens or CEP-95 for NFTs) provide on-chain names and symbols, underscoring the community’s move toward rich on-chain identification.
+
+**Metadata Fields**:
+
+- `contract_name`: A human-readable name for the contract (e.g. "Casper DEX").
+- `contract_description`: A brief description of the contract’s purpose or functionality.
+- `contract_icon_uri`: A URI (URL) pointing to an icon image for the contract (e.g. a logo).
+- `contract_project_uri`: A URI pointing to the contract’s project website or documentation.
+
+Each field is optional – contracts may provide all, some, or none of these. If a field is not applicable, it simply returns `None`. By keeping these fields on-chain and unchangeable after deployment, users and tools can rely on the metadata as the authentic self-declared identity of the contract, improving transparency and reducing reliance on off-chain sources.
+
+## Motivation
+
+[motivation]: #motivation
+
+- Establish a uniform and simple approach for contract metadata storage. 
+- Enhance contract discoverability and transparency for end-users. 
+- Simplify indexing and display logic for tools like wallets and explorers.
+
+## Specification
+
+[specification]: #specification
+
+Contracts implementing this standard must expose the following entry points:
+
+```rust
+pub trait CEP96ContractMetadata {
```

Comment:
I think mentioning `pub trait` may be confusing for smart contract developers, as there is currently no way for contract developers to define traits in the current smart contract engine.